cocos2d/shaders/CCGLProgramState.js

```diff
@@ -181,7 +180,10 @@ cc.GLProgramState.prototype = {
         }
          
         for (var i = 0; i < this._uniforms.length; ++i) {
-            this._uniforms[i].apply();
+            var uniform = this._uniforms[i];
+            if (uniform._currentBoundValue !== uniform._value) {
```

Comment:
As it's not very often that the bound value of uniform get modified, I'd like to avoid as much function call as possible in the rendering process. So I put the check inline